#cranebot:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 1000  

  # Effort Controllers - Joint Position Controllers ---------------------------------------
  bridge_joint_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: bridge_joint
    pid: {p: 500.0, i: 5, d: 150, i_clamp_min: -10000, i_clamp_max: 10000}

  trolley_joint_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: trolley_joint
    pid: {p: 500.0, i: 5, d: 150, i_clamp_min: -10000, i_clamp_max: 10000}

  cables_joint_z_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: cables_joint_z
    pid: {p: 30.927, i: 0, d: 0.0, i_clamp_min: -10000, i_clamp_max: 10000}

  platform_joint_x_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: platform_joint_x
    pid: {p: 953.2, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  platform_joint_y_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: platform_joint_y
    pid: {p: 891.5, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  platform_joint_z_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: platform_joint_z
    pid: {p: 551.3, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  platform_erb145_joint_A_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: platform_erb145_joint_A
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  erb145_link2_joint_A_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: erb145_link2_joint_A
    pid: {p: 16000, i: 0, d: 25, i_clamp_min: -10000, i_clamp_max: 10000}

  link2_erb145_joint_A_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: link2_erb145_joint_A
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  erb145_link4_joint_A_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: erb145_link4_joint_A
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  link4_erb115_joint_A_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: link4_erb115_joint_A
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  link5_gripper_joint_A_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: link5_gripper_joint_A
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  platform_erb145_joint_B_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: platform_erb145_joint_B
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  erb145_link2_joint_B_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: erb145_link2_joint_B
    pid: {p: 16000, i: 0, d: 25, i_clamp_min: -10000, i_clamp_max: 10000}

  link2_erb145_joint_B_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: link2_erb145_joint_B
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  erb145_link4_joint_B_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: erb145_link4_joint_B
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  link4_erb115_joint_B_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: link4_erb115_joint_B
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}

  link5_gripper_joint_B_effort_pos_controller:
    type: effort_controllers/JointPositionController
    joint: link5_gripper_joint_B
    pid: {p: 5000, i: 0, d: 0, i_clamp_min: -10000, i_clamp_max: 10000}